### DISCUSSION TITLE
visualizing multi_class after doing one-hot in preprocessing

### DISCUSSION THREAD

**Discussion by Nastii22** [TorchIO-project/torchio#1165] [Q&A] [answered]
Hello,
I've utilized 'tio.OneHot(3)' for pre-processing. Now, I aim to visualize the results for one of the subjects, for each channel separately. Could you please guide me on how to achieve this?
Thank you.

**Comment by romainVala** [comment, marked as answer]
Hello
For visualisation I use the torchio Image save funtion to save it as a 4D nifti file
`subject.label.save('oneHot4D.nii')` (if the subject tag of the image is "label" )
and then you can go to any nifti viewer (Slicer / itksnap / fsleyes (from FSL) / mrview (from mrtrix) / freeview (from Freesurfer)

**Reply by Nastii22** [reply]
Hi @romainVala,
Thank you for coming back to me so quickly.
So, isn't it possible to access each channel and visualize it with 'plot'?
And could you please explain the reason for using 4D?

Thank you.

**Reply by romainVala** [reply]
no the current plot from torchio does only plot the first channel

you need a 4D MRI volume because you want to look each channel separatly. In other word, you have a 4D nifti file meand you store in a single nifti file  N volumes (with N == to number of channel)

**Reply by Nastii22** [reply]
Thanks for explanation.
